src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -259,23 +258,11 @@ def execution_logs_by_dataset_name(
     """
 
     execution_logs: DefaultDict[str, List["ExecutionLog"]] = defaultdict(list)
-    SessionLocal = get_db_session()
-    db = SessionLocal()
-    # TODO future refactor for performance
-    logs = (
-        ExecutionLog.query(
-            db=db,
-        )
-        .filter(
-            ExecutionLog.privacy_request_id == self.id,
-        )
-        .order_by(
-            ExecutionLog.dataset_name,
-            ExecutionLog.updated_at.asc(),
-        )
-    )
-    for log in logs:
-        if len(execution_logs[log.dataset_name]) > EMBEDDED_EXECUTION_LOG_LIMIT:
+
+    for log in self.execution_logs.order_by(
+        ExecutionLog.dataset_name, ExecutionLog.updated_at.asc()
+    ):
+        if len(execution_logs[log.dataset_name]) > EMBEDDED_EXECUTION_LOG_LIMIT - 1:
```

Comment:
Why are logs being skipped based on the length of their name?

Reply:
@steven, it's not the length of the dataset name, this logic caps the amount of execution logs we can embed per dataset.  The execution logs are under "results" grouped by dataset.  `len(execution_logs["my-mongo-db"])` pulls the number of mongo execution logs, for example.

```json
{
    "items": [
        {
            "id": "pri_5f4feff5-fb60-4286-82bd-7e0748ce90ac",
            "created_at": "2021-10-04T17:36:32.223287+00:00",
            "started_processing_at": "2021-10-04T17:36:37.248880+00:00",
            "finished_processing_at": "2021-10-04T17:36:37.263121+00:00",
            "status": "pending",
            "results": {
                "my-mongo-db": [
                    {
                        "collection_name": "order",
                        "fields_affected": [
                            {
                                "path": "order.customer_name",
                                "field_name": "name",
                                "data_categories": [
                                    "user.provided.identifiable.name"
                                ]
                            }
                        ],
                        "message": null,
                        "action_type": "access",
                        "status": "pending",
                        "updated_at": "2021-10-05T18:24:55.570430+00:00"
                    }
                ],
                "my-postgres-db": [
                    {
                        "collection_name": "order",
                        "fields_affected": [
                            {
                                "path": "order.customer_name",
                                "field_name": "name",
                                "data_categories": [
                                    "user.provided.identifiable.name"
                                ]
                            }
                        ],
                        "message": null,
                        "action_type": "access",
                        "status": "pending",
                        "updated_at": "2021-10-05T18:24:39.953914+00:00"
                    },
                    {
                        "collection_name": "order",
                        "fields_affected": [
                            {
                                "path": "order.customer_name",
                                "field_name": "name",
                                "data_categories": [
                                    "user.provided.identifiable.name"
                                ]
                            }
                        ],
                        "message": null,
                        "action_type": "access",
                        "status": "pending",
                        "updated_at": "2021-10-05T18:24:45.240612+00:00"
                    }
                ]
            }
        }
    ],
    "total": 1,
    "page": 1,
    "size": 50
}

```

Reply:
ah, ofc.